src/sock/socket_session.hpp

```diff
@@ -56,10 +73,11 @@ class socket_session_handler;
 template <class T>
 class socket_session : public std::enable_shared_from_this<socket_session<T>>
 {
-    beast::flat_buffer buffer;                                  // used to store incoming messages
-    websocket::stream<beast::ssl_stream<beast::tcp_stream>> ws; // websocket stream used send an recieve messages
-    std::vector<T> queue;                                       // used to store messages temporarily until it is sent to the relevant party
-    socket_session_handler<T> &sess_handler;                    // handler passed to gain access to websocket events
+    beast::flat_buffer buffer;                                                   // used to store incoming messages
+    websocket::stream<beast::ssl_stream<beast::tcp_stream>> ws;                  // websocket stream used send an recieve messages
+    std::vector<T> queue;                                                        // used to store messages temporarily until it is sent to the relevant party
+    socket_session_handler<T> &sess_handler;                                     // handler passed to gain access to websocket events
+    std::unordered_map<util::SESSION_THRESHOLDS, session_thresholds> thresholds; // track down various thresholdsls
```

Comment:
It's a waste to keep a hash map to hold the set of thresholds. Because threshold types is an enum, a vector is enough for this. Threshold type enum value will be equal to vector index.